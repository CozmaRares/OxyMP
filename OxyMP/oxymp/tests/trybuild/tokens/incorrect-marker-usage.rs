// can't be a struct
#[oxymp::oxymp]
mod l2 {
    #[oxymp::Tokens]
    struct Tok;
}

// can't be a union
#[oxymp::oxymp]
mod l3 {
    #[oxymp::Tokens]
    union Tok {
        f1: u32,
        f2: f32,
    }
}

// no tokens
#[oxymp::oxymp]
mod l4 {}

// multiple tokens
#[oxymp::oxymp]
mod l5 {
    #[oxymp::Tokens]
    enum Tok1 {
        #[exact("a")]
        A,
    }

    #[oxymp::Tokens]
    enum Tok2 {
        #[exact("a")]
        A,
    }
}

fn main() {}
