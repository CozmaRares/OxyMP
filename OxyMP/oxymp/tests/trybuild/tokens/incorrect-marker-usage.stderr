error: A struct cannot be marked with #[oxymp::Tokens]. Please use enum.
 --> tests/trybuild/tokens/incorrect-marker-usage.rs:5:5
  |
5 |     struct Tok;
  |     ^^^^^^

error: A union cannot be marked with #[oxymp::Tokens]. Please use enum.
  --> tests/trybuild/tokens/incorrect-marker-usage.rs:12:5
   |
12 |     union Tok {
   |     ^^^^^

error: Module marked with #[oxymp] must contain at least one enum marked with #[oxymp::Tokens].
  --> tests/trybuild/tokens/incorrect-marker-usage.rs:20:5
   |
20 | mod l4 {}
   |     ^^

error: Module marked with #[oxymp] must contain at most one enum marked with #[oxymp::Tokens].
  --> tests/trybuild/tokens/incorrect-marker-usage.rs:24:5
   |
24 | mod l5 {
   |     ^^
